vgmstream: r1050-3448-g77cc431b -> r1702-5534-ab6139c11

vs49688 opened the pull request:
###### Motivation for this change
Updating.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

SuperSandro2000 commented on pkgs/applications/audio/vgmstream/default.nix:
```diff
@@ -1,23 +1,29 @@
-  pname   = "vgmstream";
-  version = "r1050-3448-g77cc431b";
+  pname = "vgmstream";
+  version = "r1702-5534-ab6139c11";
```
Can we just drop the commit count and just use the release tag? That would make everything much easier.

vs49688 replied:
vgmstream doesn't really have release tags. Their page says:
> Please note that our versions are simply generated every now and then from the latest commits. Automated builds should be as stable as possible.

My plan was to simply use whatever version is current whenever `nixpkgs-update` runs.

SuperSandro2000 replied:
I don't like that. This means r-ryantm creates an update every run which would be every few days. Can we just rely on the tags in the repo or manual bumps?

vs49688 replied:
We could, but that goes against the mission "to make nixpkgs the most up-to-date repository of software in the world by the most ridiculous margin possible" ;)

On a more serious note, how do we handle other fast-moving packages that don't tag releases? Is there some sort of official policy on it?

SuperSandro2000 replied:
This just creates a lot of unnecessary commits which require a lot of reviewer resources with no real benefit. We do not need to be up to date to the day or hour.

> On a more serious note, how do we handle other fast-moving packages that don't tag releases? Is there some sort of official policy on it?

They are tagged unstable-XXXX-XX-XX and bumped at most once a week.

vs49688 replied:
Hmmm, I could modify the update script to only run if it's been a week (or some interval) since the last commit. Would that be acceptable?

I'd much prefer to keep the version format as-is (to match upstream) though. At the very least, even if the version is changed to unstable-XXXX-XX-XX, I'd like to keep the version string so it can be passed through as a `#define`.

SuperSandro2000 replied:
> Hmmm, I could modify the update script to only run if it's been a week (or some interval) since the last commit. Would that be acceptable?

Yeah, we could do that to prevent excessive updates.

vs49688 replied:
Finally got around to doing this. It now won't do anything if it's been less than a week between commits.

vs49688 committed:
vgmstream: r1050-3448-g77cc431b -> unstable-2022-02-21

* Enable Speex support
* Disable the new FSB CELT support
  - It always attempts to download it, there's no option
    to use the system version.

vs49688 committed:
vgmstream: add update script
